Merge Homepage with All Available Components to the development branch

amnabbouti opened the pull request:
This pull request includes the initial setup of the Homepage component, integrating all available components together, the following components have been added and structured within the Homepage:  
```
Header
Footer
ShopGrid
ProductGrid
HeadingTxt
Input
Img
Button
UsBanner
BannerShop
FilterSortBar
LoadingSkeleton
```

The Homepage component is now fully assembled and ready for further development and styling adjustments. This merge aims to provide a foundational layout for the homepage, ensuring all key components are in place.  Changes:  
- [x] Created and structured the Homepage component.
- [x]  Imported and utilized all necessary components.
- [x]  Ensured the layout follows the intended design.
- [x]  All components have the necessary styling following the design.

**Testing:**  
- [x]  Verified that all components render correctly on the Homepage.
- [x]  Checked for any console errors or warnings.

**-Notes:**  
Further styling and functionality enhancements will be addressed in subsequent pull requests.

Joppe-VH reviewed (changes requested):
Went through it all and made some notes. 
Some things I mentioned can be merged as is if a ticket for them is made in backlog so we don't forget to update them later.
That said, I think I can make most of the changes I requested pretty quickly so if it would be of help I'm totally up for implementing them myself later today. (tuesday) At least the ones everyone agrees on.

I'll discuss this more tomorrow, going to sleep now. Just wanted to get the review done so this can be merged rather sooner than later.

P.S.: and just to clarify, I understand that it is in everyone's intrest to merge this quickly, I also know that nothing is more permanent than a temporary fix, hence the somewhat detailed review.

amnabbouti commented on src/components/Headingtxt.tsx:
```diff
@@ -0,0 +1,22 @@
+import { typographySizes } from "../utils/typographySizes.ts";
+import styles from "../scss/abstracts/typography.module.scss";
+
+const Heading: React.FunctionComponent<HeadFootProps> = ({
```
Headingtxt is used through the whole page so it's important to name them differently for clarity.

amnabbouti commented on src/components/Headingtxt.tsx:
```diff
@@ -0,0 +1,22 @@
+import { HeadFootProps } from "../types/types.ts";
```
i still prefer personally to work with a separated types file but i do understand your point, and it's a very good one, i will make sure to make the necessary changes.

amnabbouti commented on src/pages/Homepage.tsx:
good seen, will make the necessary changes.

amnabbouti commented on src/components/Header.tsx:
```diff
@@ -0,0 +1,52 @@
+}: HeadFootProps) {
+  return (
+    <header className={`${styles.header} ${className}`} {...props}>
+      <img src={PokeLogo} alt="Headerlogo" className={styles.headerLogo} />
```
good catch, changes made.

amnabbouti replied:
almost forgot but using <Link> is better than <a> tags for internal navigation in React actually.

amnabbouti commented on src/components/LoadingSkeleton.tsx:
```diff
@@ -0,0 +1,18 @@
+const LoadingSkeleton = () => {
+  return (
+    <div className={styles.skeletonGrid}>
+      {Array.from({ length: 16 }).map((_, index) => (
```
Not a huge impact here, but I’ll keep it in mind for cases where performance is more critical.

amnabbouti commented on src/components/LoadingSkeleton.tsx:
yep, definitely something to fix! Already noted in the PR—just scroll up to [this comment](https://github.com/Poke-market/poke_markt_webshop/pull/9#issue-2876255004) where I mentioned that further styling and functionality enhancements will be handled in future PRs.

amnabbouti commented on src/components/ProductCard.tsx:
```diff
@@ -0,0 +1,65 @@
+  if (text.length <= maxLength) {
+    return text;
+  }
+  return text.substring(0, maxLength) + "...";
```
The truncating was added here just for testing reasons, not sure if we gonna need that with our actual API.

amnabbouti commented on src/components/Shopgrid.tsx:
```diff
@@ -0,0 +1,125 @@
+  }, [initialData]);
+
+  // Added scroll to top when the page changes
+  useEffect(() => {
+    window.scrollTo({ top: 0, behavior: "smooth" });
+  }, [currentPage]);
```
that's a good idea, can definitely be implemented, will adress that in a separated  PR.
trello ticket is made.

amnabbouti commented on src/components/Shopgrid.tsx:
that's necessary, this one is only for testing with a Fake API.

amnabbouti commented on src/scss/partials/_variables.scss:
```diff
@@ -0,0 +1,39 @@
+$font-weight-semi-bold: 600;
+$font-weight-light: 300;
+
+// productCard
+$product-image-height: 300.96px;
+$discount-badge-size: 4.8rem;
+$discount-badge-font-size: 1.6rem;
+$details-container-bg-color: rgba(241, 241, 246, 0.74);
+$product-name-font-size: 2.4rem;
+$product-name-color: #3a3a3a;
+$product-description-font-size: 1.6rem;
+$product-description-color: #898989;
+$current-price-font-size: 2rem;
+$current-price-color: #3a3a3a;
+$original-price-font-size: 1.6rem;
+$original-price-color: #b0b0b0;
```
Fair point! Keeping it global for now since it can be used through multiple components, this will be revisited and scoped down if needed later.

amnabbouti commented on src/types/types.ts:
```diff
@@ -0,0 +1,52 @@
+  onClick?: () => void;
+};
+
+export type ButtonProps = {
```
Yeah, makes sense! Extending ComponentProps<"button"> would make it more flexible and reduce redundancy, I’ll update it in a future refactor!

amnabbouti commented on src/utils/index.ts:
This barrel file is best kept in the utils folder as we can combine it with other utility exports throughout the project.

amnabbouti commented on src/scss/partials/_variables.scss:
```diff
@@ -0,0 +1,39 @@
+@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@400;500&display=swap");
+
+// colors
+$text-color: #000000;
+$discount-color-30: #e97171;
+$background-color: #f4f5f7;
+$text-color: #000000;
+$text-color-2: #242424;
+$grey: #ccc;
+$white: #fff;
+$basic-blue: #3a76cf;
+$light-blue: #9dbbe7;
+$red: #ff5a50;
```
that would be needed if we implement dark mode i guess,  changed are made to css variables.

amnabbouti commented:
Implemented requested changes from @Joppe-VH  and cleaned up the code for better clarity. This version is prepared for a test deployment, with future updates planned for API logic and other functionalities.

amnabbouti commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Create page/homepage  and Integrating Components](https://trello.com/c/wbL2RtjM/67-create-page-homepage-and-integrating-components)

amnabbouti commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Define global color vars in css, refactor existing components to use these](https://trello.com/c/0pPgcFRh/61-define-global-color-vars-in-css-refactor-existing-components-to-use-these)
